packages/api/src/index.ts

```diff
@@ -86,10 +151,20 @@ const getApp = async () => {
 
 	app.use('/api', apiRouter);
 
+	const clientPages = ['export'];
+
 	if (runningOnAws) {
 		app.use(express.static('client'));
-		app.get('/*', (req, res) => {
-			res.sendFile(path.resolve(__dirname, 'client', 'index.html'));
+		app.get('/:page', (req, res) => {
+			if (req.params.page && !clientPages.includes(req.params.page)) {
+				res
+					.status(404)
+					.send(
+						`Endpoint not supported. Valid endpoints: /, /${clientPages.join(', /')}`,
+					);
+			}
+			const page = req.params.page ? `${req.params.page}.html` : 'index.html';
+			res.sendFile(path.resolve(__dirname, 'client', page));
```

Comment:
I thought the nextjs had routing mechanism?  What was the limitation in this case?

Reply:
It looks like nextjs deliberately generates different html files for each page for performance reasons rather than working as a single page app. https://nextjs.org/docs/app/building-your-application/deploying/static-exports